packages/@webex/plugin-meetings/src/meeting/index.ts

```diff
@@ -7401,4 +7422,22 @@ export default class Meeting extends StatelessWebexPlugin {
     // (permissionTokenExp is a epoch timestamp, not a time to live duration)
     return (permissionTokenExpValue - now) / 1000;
   }
+
+  /**
+   * Check if there is enough time left till the permission token expires
+   * If not - refresh the permission token
+   *
+   * @param {number} threshold - time in seconds
+   * @param {string} reason - reason for refreshing the permission token
+   * @returns {Promise<void>}
+   */
+  public checkAndRefreshPermissionToken(threshold: number, reason: string): Promise<void> {
+    const permissionTokenTimeLeft = this.getPermissionTokenTimeLeftInSec();
+
+    if (permissionTokenTimeLeft <= threshold) {
+      return this.refreshPermissionToken(reason);
```

Comment:
Question: looked at refreshPermissionToken, it can return without throwing error, so how do we deal that scenario?

Reply:
we just continue the join flow - refreshing the permission token is a "safety" mechanism - but if for example the permission token is fresh and not about to expire, the function will just return and the join flow just continues as normal.